fitbenchmarking/controllers/lmfit_controller.py

```diff
@@ -0,0 +1,159 @@
+                       'cobyla',
+                       'slsqp',
+                       'emcee'],
+        'general': [],
+        'simplex': ['nelder'],
+        'trust_region': ['least_squares',
+                         'trust-ncg',
+                         'trust-exact',
+                         'trust-krylov',
+                         'trust-constr',
+                         'dogleg'],
+        'levenberg-marquardt': ['leastsq'],
+        'gauss_newton': ['newton',
+                         'tnc'],
+        'bfgs': ['lbfgsb',
+                 'bfgs'],
+        'conjugate_gradient': ['cg'],
+        'steepest_descent': [],
+        'global_optimization': ['differential_evolution',
+                                'ampgo',
+                                'shgo',
+                                'dual_annealing']
+        }
+
+    jacobian_enabled_solvers = ['cg',
+                                'bfgs',
+                                'newton',
+                                'lbfgsb',
+                                'tnc',
+                                'slsqp',
+                                'dogleg',
+                                'trust-ncg',
+                                'trust-krylov',
+                                'trust-exact']
+
+    hessian_enabled_solvers = ['newton',
+                               'dogleg',
+                               'trust-constr',
+                               'trust-ncg',
+                               'trust-krylov',
+                               'trust-exact']
+
+    def __init__(self, cost_func):
+        """
+        Initialises variables used for temporary storage.
+        :param cost_func: Cost function object selected from options.
+        :type cost_func: subclass of
+                :class:`~fitbenchmarking.cost_func.base_cost_func.CostFunc`
+        """
+        super().__init__(cost_func)
+        self.support_for_bounds = True
+        self.bound_minimizers = ['dual_annealing', 'differential_evolution']
+        self.lmfit_out = None
+        self.lmfit_params = Parameters()
+
+    def lmfit_resdiuals(self, params):
+        """
+        lmfit resdiuals
+        """
+        return self.cost_func.eval_r(list(map(lambda name: params[name].value,
+                                     self.problem.param_names)))
+
+    def lmfit_jacobians(self, params):
+        """
+        lmfit jacobians
+        """
+        return self.cost_func.jac_cost(list(map(lambda name:
+                                       params[name].value,
+                                       self.problem.param_names)))
+
+    def setup(self):
+        """
+        Setup problem ready to be run with lmfit
+        """
+
+        if (self.value_ranges is None or np.any(np.isinf(self.value_ranges))) \
+           and self.minimizer in self.bound_minimizers:
+            raise MissingBoundsError(
+                    "Differential evolution requires finite bounds on all"
```

Comment:
This `MissingBoundsError` message should be dependent on the current minimizer, as it applies to both differential evolution and dual annealing